packages/engine-test-utils/src/__tests__/common-all/dnode.spec.ts

```diff
@@ -241,4 +245,202 @@ describe(`SchemaUtil tests:`, () => {
       });
     });
   });
+
+  describe("Schema match tests", () => {
+    testSchemaMatches({
+      name: "GIVEN schema anatomy example from the wiki",
+      schema: `version: 1
+schemas:
+# this will match "cli.*" notes
+- id: cli 
+  # human readable description of hierarchy
+  desc: command line interface reference
+  # add this to the domain of your schema hierarchy
+  parent: root
+  # when a schema is a namespace, it can have arbitrary children. equivalent to cli.* glob pattern
+  namespace: true 
+  children:
+    - cmd
+    - env
+# will match cli.*.env
+- id: env
+  desc: cli specific env variables
+# will match cli.*.cmd.*
+- id: cmd
+  desc: subcommands 
+  namespace: true`,
+      checks: {
+        "cli.fd": "cli",
+        "cli.fd.bar.usage": false,
+        "cli.fd.cmd.usage": "cmd",
+        "cli.fd.env": "env",
+        "foo.bar": false,
+      },
+    });
+
+    testSchemaMatches({
```

Comment:
I'm guilty of introducing the `testWithEngine` pattern but have reconsidered it because:
- it's implementation specific (assumes jest, doesn't work with mocha test)
- not easily compsable (doesn't let us use modifiers like `.only` easily)

Can we use a utility that doesn't actually run a test but just sets up the test case?